Entity/CustomItem.php

```diff
@@ -611,4 +620,32 @@ public function getRelationsByType(string $entityType): Collection
                 return new ArrayCollection([]);
         }
     }
+
+    public function getUniqueHash(): ?string
+    {
+        return $this->uniqueHash;
+    }
+
+    public function setUniqueHash(?string $uniqueHash): void
+    {
+        $this->uniqueHash = $uniqueHash;
+    }
+
+    public function createUniqueHash(): ?string
```

Comment:
Thank you for the changes.

Right now, I would say this method violates Single Responsibility and Open Closed. I think we should extract it into a service or change the method to `createUniqueHash(): void` and have it set the hash instead of returning it.

Reply:
That's a good point. The way I had implemented it in the initial stages of the code was that I had called the `setUniqueHash()` method within this one. So instead of returning the unique hash, it was setting it directly. I changed that implementation because I felt "creating" and "setting" are 2 different functions and if the hash is set in this method, it won't be exactly true to its name.

Reply:
Sorry I meant `updateUniqueHash(): void`.

I agree that get and set do two different things and `setUniqueHash()` wouldn't be the write approach.

There is also the concept of pure functions. `createUniqueHash(): ?string` is not pure because it relays on the internal state. `updateUniqueHash(): void` is pure. Instead of thinking about things as simply as setting or getting a value, I view entities as persisted state data. All the methods on it should get state data or manipulate the state in some way. If they do neither they don't belong in the entity.

Making this a service with a method like `createUniqueHash(ExtractedInterface $entity): void` is the most SOLID but it will come with a small performance hit. I don't think `createUniqueHash` is complex and the need overwrite/replace it is unlikely IMO, YAGNI, so it's not worth the hit in performance. 

Not valid here but food for thought. A third way to approach things like this if `customObject` was a value or an immutable object is to do it on `setCustomObject`. `setCustomObject` would calls a private method like  `createUniqueHash(): ?string`. This ensures that the `getUniqueHash` is `O(1)` and always the correct value. If `createUniqueHash(): ?string` was expensive and set multiple times before a get, it is optimized by using a dirty flag so that set is `O(1)` and get is `O(1)` when clean and `O(createUniqueHash)` when dirty.

Reply:
I don't think `setCustomObject` would be the right place to do it, `setCustomItem`, maybe. But again, `setCustomItem` is being used for an entirely different purpose here. I looked through the code once again and I could find any other function which could be a suitable candidate for calling the said private method.

Now that I see it from your POV, I really like the `updateUniqueHash(): void` idea you suggested. It might prove to be a simple yet effective solution to implement that. I just feel a more apt name would be `createAndSetUniqueHash(): void`. This method would create AND set the unique hash. This way we eliminate the dependance on internal state, while not disturbing the original setter function. 

Are we on the same page here?

Reply:
Yup i'd suggest `updateUniqueHash(): void`.

I went on a tangent explaining when doing something like this in a setter would make sense. Your intuition is 100% correct, on why `setCustomObject` is not good here.

Reply:
Thanks! I have made the changes, have a look, please